helix-core/src/main/java/org/apache/helix/manager/zk/client/HelixZkClient.java

```diff
@@ -35,9 +36,47 @@
 
   void unsubscribeDataChanges(String path, IZkDataListener listener);
 
-  void subscribeStateChanges(final IZkStateListener listener);
+  /*
+   * This is for backwards compatibility.
+   *
+   * TODO: remove below default implementation when getting rid of I0Itec in the new zk client.
+   */
+  default void subscribeStateChanges(final IZkStateListener listener) {
```

Comment:
For a standard interface comment, we need to comment on how this one is supposed to be implemented. Please put the todo and backward compatibility after that.